src/components/panel/panel.js

```diff
@@ -1663,6 +1809,42 @@ MdPanelRef.prototype._done = function(callback, self) {
 };
 
 
+/**
+ * Adds a panel to a group if the panel does not already exist within the group.
+ * @param {string} groupName The name of the group.
+ */
+MdPanelRef.prototype.addToGroup = function(groupName) {
+  if (!this._$mdPanel._groups[groupName]) {
+    this._$mdPanel.newPanelGroup(groupName);
+  }
+
+  var group = this._$mdPanel._groups[groupName];
+
+  group.panels.push(this);
+};
+
+
+
+/**
+ * Removes a panel from a group if the panel exists within that group.
+ * @param {string} groupName The name of the group.
+ */
+MdPanelRef.prototype.removeFromGroup = function(groupName) {
+  if (!this._$mdPanel._groups[groupName]) {
+    throw new Error('The group does not exist.');
+  }
+
+  var group = this._$mdPanel._groups[groupName];
+  var index = group.panels.indexOf(this);
+
+  if (index > -1) {
+    group.panels.splice(index, 1);
+  } else {
+    throw new Error('The panel does not exist within the group.');
```

Comment:
I'd argue that if the panel wasn't in the group, then the remove action was successful so we shouldn't throw an error here.